docs/scripting/language/Style.md

```diff
@@ -0,0 +1,216 @@
+---
+id: Style
+title: "Pawn Style Guide"
+description: A short guide on the generally accepted naming conventions and other aspects of Pawn source code to aid easier communication of intent and streamline debugging and sharing of code.
+---
+
+This document is a short guide on the generally accepted naming conventions and other aspects of Pawn source code to aid easier communication of intent and streamline debugging and sharing of code.
+
+See also:
+
+- [Modern Pawn](https://github.com/Southclaws/sampctl/wiki/Modern-Pawn)
+- [Pawn Package](https://github.com/Southclaws/sampctl/wiki/Packages)
+
+## Terminology
+
+### Modules
+
+A **Module** is a single Pawn source file, with an `.inc` extension.
+
+Some internal functions can not use `static` if they are used across different modules in the same package. When this occurs, functions must use a name style to indicate that they are _private_. See the [function style](#functions) section for details.
+
+Modules should avoid exposing global variables. Functions must be used instead with, where appropriate, bounds checking and validity checks.
+
+### Statement
+
+(TODO)
+
+### Compound Statement
+
+(TODO)
+
+### Expression
+
+(TODO)
+
+## Guidelines
+
+### Braces
+
+Allman braces are preferred:
+
+```pawn
+function()
+{
+    //
+}
+```
+
+However, if you can't shake the muscle memory, K&R braces are also valid Pawn:
+
+```pawn
+function()
+{
+    //
+}
+```
+
+### Switches
+
+Switches must use two indent levels, one for the `switch` block and another for each `case` statement or compound statement.
+
+```pawn
+switch (variable)
+{
+    case 0:
+        return 0;
+    case 1:
+        return 1;
+    case 2:
+        return 2;
+    default:
+        return -1;
+}
+```
+
+```pawn
+switch (variable)
+{
+    case 0:
+    {
+        // code...
+        return 0;
+    }
+    case 1:
+    {
+        // code...
+        return 1;
+    }
+    case 2:
+    {
+        // code...
+        return 2;
+    }
+    default:
+    {
+        // code...
+        return -1;
+    }
+}
+```
+
+### Compound Statements (Blocks)
+
+Blocks must always use braces, even if only a single line of code exists within a block. This applies to all levels including functions.
+
+```pawn
+func()
+{
+    singleLineExpr();
+}
+```
+
+```pawn
+func()
+{
+    if()
+    {
+        singleLineExpr();
+    }
+}
+```
+
+```pawn
+func()
+{
+    if()
+    {
+        singleLineExpr();
+    }
+    else if()
+    {
+        //
+    }
+    else
+    {
+        //
+    }
+}
+```
+
+### Naming
+
+#### Functions
+
+Functions must be named with either:
+
+- `PascalCase` if they are used externally
+- `camelCase` if `static` or only used internally within the same package
```

Comment:
How common is this in Pawn? Most internal functions I see are prefixed with `_`

Reply:
I prefer PascalCase for API functions, and camelCase for static functions too. So I can agree with this.

Reply:
_ makes sense too, I've seen this and I think I've even used this convention.

Reply:
I am against `camelCase` for any kind of function name, however any global variable, constant or function parameter should be in `camelCase`.

Reply:
I *think* using _ is actually considered a bad practice, that's adding to convention list and actually a way to confuse people who come from other languages.

In this case I guess it's just something like what others are probably doing it wrong in Javascript:
```
Do not use _ *underbar* as the first or last character of a name. It is sometimes intended to indicate privacy, but it does not actually provide privacy. If privacy is important, use closure. Avoid conventions that demonstrate a lack of competence.
```
Source: https://www.crockford.com/code.html

if it should be private, and compiler can handle it if you use `static` keyword, then what's against you?

Reply:
The case for when you can't use `static` is when there's a folder of pawn source files and your function is shared within those but not externally.

But I may just remove this to get the thing merged faster and we can worry about that later in another PR.